frontend/src/components/home/measurements/client/ClientsAndProjectsEvolution.tsx

```diff
@@ -0,0 +1,142 @@
+
+  while (interim.startOf("month").isSameOrBefore(to.endOf("month"))) {
+    timeValues.push(interim.clone());
+    interim.add(1, "month");
+  }
+
+  const dataSet: clientChartData[] = timeValues.map((date) => {
+    return {
+      year: date.year(),
+      monthIndex: date.month(),
+      month: date.format("MMM"),
+      clients: null,
+      clientsWithProjects: null,
+    };
+  });
+
+  const clientsWithProject = models.clients.map((client) => {
+    return {
+      detail: client,
+      projects: models.projects.filter(
+        (project) => project.client.clientId === client._id
+      ),
+    };
+  });
+
+  let count: number = 0;
+  dataSet.forEach((entry) => {
+    let amount: number = models.clients
+      .filter(
+        (client) => moment.utc(client.audit.createdOn).year() === entry.year
+      )
+      .filter(
+        (client) =>
+          moment.utc(client.audit.createdOn).month() === entry.monthIndex
+      ).length;
+    count += amount;
+    entry.clients = count;
+  });
+
+  /* Here I check if a client has a project with a start and enddate 
+  that is valid for the month and year pairs in the dataset and count them up for the year and month pairs. */
+
+  dataSet.forEach((entry) => {
+    let amount: number = clientsWithProject.filter(
+      (client) =>
+        client.projects.length > 0 &&
+        client.projects.some((project) =>
+          moment([entry.year, entry.monthIndex]).isBetween(
+            project?.startDate,
+            project?.endDate
+          )
+        )
+    ).length;
+    entry.clientsWithProjects = amount;
+  });
+
+  /* For the clients with projects column in the list 
+  I want to check and count the unique clients with a valid contract in the whole period -> I store this
+  separately inside allClientsWithProjects */
+
+  let allClientsWithProjects = clientsWithProject.filter(
+    (client) =>
+      client.projects.length > 0 &&
+      client.projects.some(
+        (project) =>
+          project?.startDate.isSameOrBefore(to) &&
+          (project.endDate === undefined || project.endDate.isSameOrAfter(from))
+      )
+  ).length;
+
+  return (
+    <>
+      <Container>
+        <Row>
+          <Link to={`/clients`}>
+            <h5>
+              {t(
+                "measurements.clientSection.clientsAndProjectsEvolution.title"
+              )}
+            </h5>
+          </Link>
+          <ClientsAndProjectsEvolutionList
+            from={from}
+            to={to}
+            clients={dataSet[dataSet.length - 1].clients || 0}
+            clientsWithProject={allClientsWithProjects}
+          />
+        </Row>
+        <Row>
+          <Col className='align-items-bottom'>
+            <PeriodPicker
+              from={from}
+              to={to}
+              fromChange={(value) => {
+                value && value < to && setFrom(value);
+              }}
+              toChange={(value) => {
+                value && value < to && setTo(value);
+              }}
```

Comment:
Dit is dan allemaal logica die in het component zelf zou komen

Reply:
Met beide abstracties:
- from/to useState --> wordt 1 useState
- from/to params --> wordt 1 param
- fromChange/toChange --> wordt 1 callback